src/Lorentz/Contracts/BaseDAO/FA2.hs

```diff
@@ -5,29 +5,255 @@ module Lorentz.Contracts.BaseDAO.FA2
    ( transfer
    , balanceOf
    , tokenMetadataRegistry
-   , permissionsDescriptor
    , updateOperators
+   , defaultPermissionsDescriptor
    ) where
 
 import Lorentz
 import Lorentz.Contracts.BaseDAO.Types
 import Lorentz.Contracts.Spec.FA2Interface
 
--- | TODO: Implement missing methods
+defaultPermissionsDescriptor :: PermissionsDescriptor
+defaultPermissionsDescriptor = PermissionsDescriptor
+  { pdOperator = OwnerOrOperatorTransfer
+  , pdReceiver = OwnerNoHook
+  , pdSender = OwnerNoHook
+  , pdCustom = Nothing
+  }
+
 transfer :: Entrypoint TransferParams Storage
-transfer = doNothing
+transfer = do
+  iter transferItem
+  nil; pair
+  where
+    transferItem :: forall s. TransferItem & Storage & s :-> Storage & s
```

Comment:
I believe it is better to use explicit stack here.

```haskell
transferItem :: forall s. '[TransferItem, Storage] :-> '[Storage]
```

The reason being that, using `TransferItem & Storage & s` is allowed when the stack is,  for example: `'[TransferItem, Storage, Storage]`.

This result in `[Storage, Storage]` where one is different from another, and can be hard to track down if we somehow use wrong `Storage` down later on.

 Using `'[TransferItem, Storage] :-> '[Storage]` guarantee that we get only 1 storage left, so we are sure that this is the `Storage` with updated state.

I think this applies to __all functions that modify storage value__.

I believe using `TransferItem & Storage & s` is more fitted in a case where we simply want to check some condition and not updating storage. We do not care about the storage value in this case.

Reply:
How to apply it correctly? I mean, if I update functions with that, I get a bunch of errors, as compiler can't deduce stack correctly

Reply:
@PruStephan Which function that it does not work? I tried with `transferItem` , it works fine for me:

```
    transferItem :: '[TransferItem, Storage] :-> '[Storage]
    transferItem = do
      swap
      getField #sAdmin
      Lorentz.sender
      eq
      dig @2
      getField #tiTxs
      swap
      getField #tiFrom
      swap
      drop @TransferItem
      dug @2
      iter transferOne
      dropN @2
```

Reply:
Change your `transferOne` type to this: 

```haskell
transferOne :: '[TransferDestination, Bool, Address, Storage] :-> '[Bool, Address, Storage]
```

Then you can inspect `debitFrom` and `creditTo` type.

```haskell
transferOne = do
      ... 
      stackType @'[Storage, Address, (TokenId, Natural), Address, (TokenId, Natural), Address, Bool];
      debitFrom;

      stackType @'[Storage, Address, (TokenId, Natural), Address, Bool];
      creditTo;

      stackType @'[Storage, Address, Bool];
```

As you can see, there are more items at the end, you have to explicitly list all of them:

```haskell
debitFrom
  ::  '[Storage, Address, (TokenId, Natural), Address, (TokenId, Natural), Address, Bool]
  :-> '[Storage, Address, (TokenId, Natural), Address, Bool]
```

Same applies to `creditTo`.

I know it may be not pretty this way, but we can optimize it later on if necessary.

Reply:
Now that I think about it, we shouldn't use explicit stack on `debitFrom` and `creditTo` since they are meant to be reusable. 

So forget about my above comment. Only use explicit stack on function that modify storage and is not meant to be resuable.

Reply:
Updated